### PR TITLE
DOCS-7105 Email Connector 1.2 Reference

### PR DESCRIPTION
**Pull request by valkyrie69**

### CODE REVIEW

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 1]
```diff
@@ -289,388 +274,370 @@ The returned connection is secured by TLS.
...
 
 [[markAsDeleted]]
 === Mark As Deleted
-
 `<email:mark-as-deleted>`
 
-Marks an incoming email as DELETED, this way the marked email(s) are scheduled for deletion when the folder closes. This means that the email is not physically eliminated from the mailbox folder, but its state changes.
 
-All DELETED marked emails are going to be eliminated from the mailbox when one of IMAPOperations#expungeFolder(MailboxConnection, String) or IMAPOperations#delete(MailboxConnection, String, long) is executed.
+Marks an incoming email as `DELETED`. Emails that are marked for deletion are scheduled for deletion when the folder closes, which means that the email is not physically eliminated from the mailbox folder, but that its state changes.  All emails that are marked as `DELETED` are eliminated from the mailbox when one of `IMAPOperations#expungeFolder(MailboxConnection, String)` or `IMAPOperations#delete(MailboxConnection, String, long)` is executed. This operation targets a single email.
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc]:
```suggestion
Marks an incoming email as `DELETED`. Emails that are marked for deletion are scheduled for deletion when the folder closes, which means that the email is not physically eliminated from the mailbox folder, but its state changes.  All emails that are marked as `DELETED` are eliminated from the mailbox when one of `IMAPOperations#expungeFolder(MailboxConnection, String)` or `IMAPOperations#delete(MailboxConnection, String, long)` is executed. This operation targets a single email.
```

**Review by ndinu — APPROVED** [review]

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 2]
```diff
@@ -289,388 +274,370 @@ The returned connection is secured by TLS.
...
 [[send]]
 === Send
-
 `<email:send>`
 
 
-Sends an email message. The message will be sent to all recipient toAddresses, ccAddresses, bccAddresses specified in the message.
-
-The content of the message aims to be some type of text (text/plan, text/html) and it's composed by the body and its content type. If no content is specified then the incoming payload it's going to be converted into plain text if possible.
+This operation sends an email message. The message is sent to all recipient  `To`, `CC`, and `BCC` addresses specified in the message.  The content of the message is some type of text (`text/plain`, `text/html`) and is composed by the body and its content type. If no content is specified then the incoming payload is converted into plain text if possible.
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc]:
```suggestion
This operation sends an email message. The message is sent to all recipient  `To`, `CC`, and `BCC` addresses specified in the message.  The content of the message is some type of text (`text/plain`, `text/html`) and is composed of the body and its content type. If no content is specified then the incoming payload is converted into plain text if possible.
```

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 3]
```diff
@@ -289,388 +274,370 @@ The returned connection is secured by TLS.
...
-| Encoding a| String |  The character encoding of the body. If it is configured, it overrides the one inferred from the body. |  |
-| Content Transfer Encoding a| String |  Encoding used to indicate the type of transformation that has been used in order to represent the body in an acceptable manner for transport. The value is case insensitive.
-
-Known encodings: BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY |  Base64 |
-| Attachments a| Object |  The attachments for an Email, that will be sent along the email body. |  |
-| Content Transfer Encoding a| String |  Encoding used to indicate the type of transformation that has been used in order to represent the body in an acceptable manner for transport. The value is case insensitive.
-
-Known encodings: BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY |  Base64 |
+| Configuration | String | The name of the configuration to use | | x
+| From Address a| String |  The "From" address of the message sender. sender address. If not set, it defaults to the "From" address specified in the config. |  |
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc]:
```suggestion
| From Address a| String |  The "From" address of the message sender. If not set, it defaults to the "From" address specified in the config. |  | 
```

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 4]
```diff
@@ -289,388 +274,370 @@ The returned connection is secured by TLS.
...
-
-Known encodings: BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY |  Base64 |
+| Configuration | String | The name of the configuration to use | | x
+| From Address a| String |  The "From" address of the message sender. sender address. If not set, it defaults to the "From" address specified in the config. |  | 
+| To Addresses a| Array of String |  The recipient addresses of "To" (primary) type |  | x
+| Cc Addresses a| Array of String |  The recipient addresses of "CC" (carbon copy) type |  | 
+| Bcc Addresses a| Array of String |  The recipient addresses of "BCC" (blind carbon copy) type |  | 
+| Reply To Addresses a| Array of String |  The email addresses to which to reply to this email |  | 
+| Subject a| String |  The subject of the email |  [No Subject] | 
+| Headers a| Object |  The headers that this email carry |  |
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc]:
```suggestion
| Headers a| Object |  The headers that this email carries |  | 
```

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 5]
```diff
@@ -289,388 +274,370 @@ The returned connection is secured by TLS.
...
+| Subject a| String |  The subject of the email |  [No Subject] | 
+| Headers a| Object |  The headers that this email carry |  | 
+| Content a| Binary |  Text body of the message, which can be in any format |  `#[payload]` | 
+| ContentType a| String |  Content Type of the body text. Example: "text/plain" |  | 
+| Encoding a| String |  The character encoding of the body. If it is configured, it overrides the one inferred from the body. |  | 
+| Content Transfer Encoding a| String |  Encoding used to indicate the type of transformation that has been used to represent the body in an acceptable manner for transport. The value is case insensitive.  Known encodings:  BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY  |  `Base64` | 
+| Attachments a| Object |  The attachments that are sent along with the email body |  | 
+| Content Transfer Encoding a| String |  Encoding used to indicate the type of transformation that is used to represent the body in an acceptable manner for transport. The value is case insensitive.  Known encodings:  BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY  |  `Base64` |
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc:533]: Is this a repeat of line 531?

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 6]
```diff
@@ -289,388 +274,370 @@ The returned connection is secured by TLS.
...
-| Imap Matcher a| <<imap-matcher>> |  A matcher to filter emails retrieved by this polling source. For default already read emails will be filtered. |  |
-| Primary Node Only a| Boolean |  Whether this source should only be executed on the primary node when runnning in Cluster |  |
-| Scheduling Strategy a| scheduling-strategy |  Configures the scheduler that triggers the polling |  | *x*
-| Redelivery Policy a| <<RedeliveryPolicy>> |  Defines a policy for processing the redelivery of the same message |  |
+| Configuration | String | The name of the configuration to use. | | x
+| Folder a| String |  The name of the folder to poll emails from.  |  `INBOX` | 
+| Delete After Retrieve a| Boolean |  Enables the deletion of the polled emails after being retrieved. This is disabled by default. | `false` | 
+| Enable Watermark a| Boolean |  If watermark should be applied to the polled emails or not. Default to true. |  true | 
+| Imap Matcher a| <<imap-matcher>> |  A matcher to filter emails retrieved by this polling source. For default already read emails will be filtered. |  |
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc]:
```suggestion
| Imap Matcher a| <<imap-matcher>> |  A matcher to filter emails retrieved by this polling source. By default already read emails will be filtered. |  | 
```

File: modules/ROOT/pages/email/email-documentation.adoc [hunk 7]
```diff
@@ -697,107 +664,107 @@ Retrieves all the emails from an POP3 mailbox folder.
...
 | Initialisation Policy a| Enumeration, one of:
 
 ** INITIALISE_NONE
 ** INITIALISE_ONE
-** INITIALISE_ALL | Determines how components in a pool should be initialized. The possible values are: INITIALISE_NONE (will not load any components into the pool on startup), INITIALISE_ONE (will load one initial component into the pool on startup), or INITIALISE_ALL (will load all components in the pool on startup) |  |
-| Disabled a| Boolean | Whether pooling should be disabled |  |
+** INITIALISE_ALL | Determines how components in a pool should be initialized. The possible values are: `INITIALISE_NONE` (will not load any components into the pool on startup), `INITIALISE_ONE` (loads one initial component into the pool on startup), or `INITIALISE_ALL` (loads all components in the pool on startup) |  |
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/email/email-documentation.adoc]:
```suggestion
** INITIALISE_ALL | Determines how components in a pool should be initialized. The possible values are: `INITIALISE_NONE` (loads no components into the pool on startup), `INITIALISE_ONE` (loads one initial component into the pool on startup), or `INITIALISE_ALL` (loads all components in the pool on startup) |  | 
```

**Review by wwarner-salesforce — APPROVED** [review]
Great to see all those small changes with big clarity improvements! I left a few comments.

File: modules/ROOT/pages/email/email-documentation.adoc (hunk 5 above)
Comment (valkyrie69) [reply modules/ROOT/pages/email/email-documentation.adoc:533]: Looks like it!

**Pull request merged**